homeassistant/components/obihai/sensor.py

```diff
@@ -48,6 +48,14 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
         _LOGGER.error("Invalid credentials")
         return
 
+    def _reboot(call):
+        """Send API call to reboot device."""
+        success = pyobihai.call_reboot()
+        if not success:
+            _LOGGER.error("Failed to reboot")
+
+    hass.services.register("obihai", "reboot", _reboot)
```

Comment:
When services are defined in platforms it should use the entity service helper <https://developers.home-assistant.io/docs/dev_101_services#entity-services>

Reply:
I tried to do that in my earliest attempts at defining the service, but I did not get it working.  I don't have what I coded or the error messages, but I believe that entity_platform.current_platform.get() was None.  Perhaps because no entities/devices are created in in the obihai integration.  I could try again if I find the time.

Reply:
I have working code with the entity service helper now.